update-tests-yaml-files

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash -e
+#
+#  Update or add one or more key:value pair in tests.yaml for all charms in charms.txt
+#  Then, propose gerrit reviews.
+#
+#  e.g. reset_timeout is now default 180s for bundletester
+#  ./update-tests-yaml-files master --kvs reset_timeout:600 will update or add
+#  this value in the first tests.yaml it finds in each charm.
+# 
+#  Note there should only be one tests.yaml per charm.
+#
+
+NO_REVIEW="True"
+
+charms="$(cat charms.txt)"
+src_charms="$(cat source-charms.txt)"
+basedir="$(pwd)"
+branch="$1"
+
+gerrit_topic="update-tests-yaml-files"
+commit_msg_template="$basedir/commit-message-tests-yaml.template"
+commit_msg_file="$basedir/commit-message-tests-yaml.txt"
+
+all_params="$@"
+if [[ "$all_params" == *--amend* ]]; then
+  AMEND="True"
+fi
+if [[ "$all_params" == *--noreview* ]]; then
+  NO_REVIEW="True"
+fi
+
+usage="usage: update-tests-yaml-files <master||stable/nn.nn> --kvs key1:value key2:value
+
+IMPORTANT: Ensure --kvs is the last param on the command line.
+
+USAGE EXAMPLES
+
+Clone repos, check out the master branch, make changes and
+submit gerrit reviews:
+  ./update-tests-yaml-files master --kvs key1:value, key2:value
+
+Clone repos, check out the master branch, make changes
+but do not actually submit a gerrit review:
+  ./update-tests-yaml-files master --noreview --kvs key1:value, key2:value
+
+Re-use local checkout, amend commits and add patchset to
+the existing gerrit review:
+  ./update-tests-yaml-files master --amend --kvs key1:value, key2:value
+
+Re-use local checkout, amend commits for adding to the review,
+but do not actually submit a gerrit review:
+  ./update-tests-yaml-files master --amend --noreview --kvs key1:value, key2:value
+"
+
+keys=`echo $@|awk -F '--kvs ' '{print $2}'`
+if [[ "$keys" == "" ]] 
+then
+        echo -e $usage
+        exit 1
+fi
+
+if [ -z "$branch" ]; then
+    echo -e "$usage"
+    exit 1
+fi
+
+# Expect user to have git config ready for gerrit use
+git config --get gitreview.username || ( echo " ! Not set: gitreview.username git config option"; echo -e "$usage"; exit 1 )
+
+commit_msg="$(cat $commit_msg_template||:)"
+if [ -z "$commit_msg" ]; then
+    echo " ! $commit_msg_file not found or empty."
+    exit 1
+fi
+
+
+function git_get(){
+  (
+  if [[ "${AMEND^^}" != "TRUE" ]] && [[ ! -d $2 ]]; then
+    echo " + Clone $1 -> $2"
+    git clone $1 $2
+    cd $2
+    git checkout $3
+  elif [[ "${AMEND^^}" != "TRUE" ]] && [[ -d $2 ]]; then
+    echo " ! Dir exists: $2.  Consider running 'make clean' or using --amend."
+    exit 1
+  else
+    echo " . Re-using checkout dir $2"
+    cd $2
+    git branch -v
+  fi
+  )
+}
+
+
+function git_review(){
+  if [ "${NO_REVIEW^^}" == "TRUE" ]; then
+    echo " . Not proposing gerrit (dry run)."
+  else
+    echo " . Submitting gerrit review for $charm"
+    git review
+  fi
+}
+
+
+function tests_yaml_kvs(){
+echo ' . Updating keys:values '
+rm ${commit_msg_file}
```

Comment:
Handling the commit message file should not be in this function. Either out in main or in its own function. This might accidentally get the file committed in the repo.